src/Factory.php

```diff
@@ -350,33 +352,40 @@ private function normalizeAttribute(mixed $value, ?string $name = null)
             $value = $value->withoutPersisting();
         }
 
-        // Check if the attribute is cascade persist
-        if (self::configuration()->hasManagerRegistry()) {
-            $ownedRelationshipField = $this->ownedRelationshipField($name, $value);
+        if (!self::configuration()->hasManagerRegistry()) {
+            return $value->create()->object();
+        }
 
-            if (null !== $ownedRelationshipField) {
-                $cascadePersist = $this->hasCascadePersist(null, $ownedRelationshipField);
-            } else {
-                $isCollection = false;
-                $relationshipField = $this->inverseRelationshipField($name, $value, $isCollection);
-                $cascadePersist = $this->hasCascadePersist($value, $relationshipField);
+        $objectManager = self::configuration()->objectManagerFor($this->class);
 
-                if ($this->isPersisting() && null !== $relationshipField && false === $cascadePersist) {
-                    $this->afterPersist[] = static function(Proxy $proxy) use ($value, $relationshipField, $isCollection): void {
-                        $value->create([$relationshipField => $isCollection ? [$proxy] : $proxy]);
-                        $proxy->refresh();
-                    };
+        if (!$objectManager instanceof EntityManagerInterface || $objectManager->getClassMetadata($value->class)->isEmbeddedClass) {
+            return $value->create()->object();
+        }
 
-                    // creation delegated to afterPersist event - return null here
-                    return self::NULL_VALUE;
-                }
-            }
+        $relationshipMetadata = self::getRelationshipMetadata($objectManager, $this->class, $name);
```

Comment:
"fun" fact: I've actually refactor all this with REALLY knowing **WHY** we are doing this

(I don't know if it is really a "fun" fact :sweat_smile: )